src/collector/top_collector.rs

```diff
@@ -175,6 +180,26 @@ impl<T: PartialOrd + Clone> TopSegmentCollector<T> {
     }
 }
 
+fn stable_sort<T, D>(a: &ComparableDoc<T, D>, b: &ComparableDoc<T, D>) -> Ordering
+where
+    T: PartialOrd,
+    D: PartialOrd,
+{
+    // Reversed since our ComparableDoc reverses the ordering (to
+    // make BinaryHeap work as a min-heap).
+    let by_feature = b
+        .feature
+        .partial_cmp(&a.feature)
+        .unwrap_or_else(|| Ordering::Equal);
+
+    let lazy_by_doc_address = || a.doc.partial_cmp(&b.doc).unwrap_or_else(|| Ordering::Equal);
```

Comment:
`unwrap_or(Ordering::Equal)`..

The closure version is only useful if you do computation in it.

Reply:
fixed